xof/blake2xb/blake.go

```diff
@@ -61,6 +66,11 @@ func (x *xof) Reseed() {
 	x.impl = y.(*xof).impl
 }
 
+func (x *xof) Reset() {
+	y := New(x.seed)
+	x.impl = y.(*xof).impl
```

Comment:
This is making too many useless allocations.
Please, refactor `New` to use `Reset` after having done the `blake2b.NewXOF` step,  and use the fact that `impl` has its own `Reset` function if you can.

Reply:
Should be fixed now !